packages/merge-styles/src/ShadowDomStylesheet.ts

```diff
@@ -0,0 +1,238 @@
+/* eslint no-restricted-globals: 0 */
+import { InjectionMode, STYLESHEET_SETTING, Stylesheet } from './Stylesheet';
+import { DEFAULT_SHADOW_CONFIG, GLOBAL_STYLESHEET_KEY, SHADOW_DOM_STYLESHEET_SETTING } from './shadowConfig';
+
+import type {
+  ExtendedCSSStyleSheet,
+  ISerializedStylesheet,
+  IStyleSheetConfig,
+  WindowWithMergeStyles,
+} from './Stylesheet';
+import type { ShadowConfig } from './shadowConfig';
+
+export const SUPPORTS_CONSTRUCTABLE_STYLESHEETS =
+  typeof document !== 'undefined' && Array.isArray(document.adoptedStyleSheets) && 'replace' in CSSStyleSheet.prototype;
+
+let supportsModifyingAdoptedStyleSheets = false;
+
+if (SUPPORTS_CONSTRUCTABLE_STYLESHEETS) {
+  try {
+    document.adoptedStyleSheets.push();
+    supportsModifyingAdoptedStyleSheets = true;
+  } catch (e) {
+    supportsModifyingAdoptedStyleSheets = false;
+  }
+}
+
+export const SUPPORTS_MODIFYING_ADOPTED_STYLESHEETS = supportsModifyingAdoptedStyleSheets;
+
+let _stylesheet: ShadowDomStylesheet | undefined;
+
+let _global: WindowWithMergeStyles = {};
+
+// Grab window.
+try {
+  // Why the cast?
+  // if compiled/type checked in same program with `@fluentui/font-icons-mdl2` which extends `Window` on global
+  // ( check packages/font-icons-mdl2/src/index.ts ) the definitions don't match! Thus the need of this extra assertion
+  _global = (window || {}) as WindowWithMergeStyles;
```

Comment:
I think this will break SSR
```suggestion
  _global = (typeof window === 'undefined' ? window : {}) as WindowWithMergeStyles;
```

Reply:
I'd prefer to leave that as the [current code does not use `typeof`](https://github.com/microsoft/fluentui/blob/master/packages/merge-styles/src/Stylesheet.ts#L113) and [all the v8 controls are currently SSR tested](https://github.com/microsoft/fluentui/blob/master/apps/ssr-tests/test/test.js#L36) (I know because I _have_ broken SSR 🙂)

Reply:
sounds good, I'll resolve the comments!